packages/core/src/utils/retry.ts

```diff
@@ -132,13 +162,17 @@ export async function retryWithBackoff<T>(
       ) {
         const jitter = currentDelay * 0.3 * (Math.random() * 2 - 1);
         const delayWithJitter = Math.max(0, currentDelay + jitter);
-        await delay(delayWithJitter);
+        await delay(delayWithJitter, signal);
         currentDelay = Math.min(maxDelayMs, currentDelay * 2);
         continue;
       }
 
       return result;
     } catch (error) {
+      if (error instanceof Error && error.name === 'AbortError') {
+        throw error;
```

Comment:
Does it make sense to throw here?

Reply:
It avoids a lot of the complicated logic below which makes this clearer but i don't think its strictly necessary